### PR TITLE
Ensure that Gacela is compatible with PHP 8.1

### PR DESCRIPTION
**Pull request by Chemaclass**
## 🔖 Changes

- Add composer cache on GitHub CI.
- Add PHP 8.1 to the `matrix.php` when running the tests.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,16 +11,29 @@ jobs:
   coding-guidelines:
     name: Coding Guidelines
     runs-on: ubuntu-latest
-
+    strategy:
+      fail-fast: false
+      matrix:
+        php: ['7.4']
```
Comment (JesusValera) [inline .github/workflows/ci.yml:17]: Only 7.4 here?

Comment (Chemaclass) [reply .github/workflows/ci.yml:17]: Yes, because it's to run the coding-guidelines (php-cs-fixer), I don't think we have to run it for all php versions.

Comment (Chemaclass) [reply .github/workflows/ci.yml]: I removed the matrix strategy for non tests jobs.

Comment (JesusValera) [reply .github/workflows/ci.yml]: It makes sense 👌🏼👌🏼

**Review by JesusValera — APPROVED** [review]

**Pull request merged**